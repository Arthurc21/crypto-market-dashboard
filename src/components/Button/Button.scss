.button {
	width: 120px;
	height: 30px;
	background-color: #f7fbfc;
	border: 1px solid #bbbbbb;
    border-radius: 4px;
	cursor: pointer;
	outline: none;
	display: inline-block;
	vertical-align: middle;

    &:hover {
        border-color: #777777;
        box-shadow: 0 0 10px 2px #888888;
    }

	&:disabled {
		opacity: 50%;
        //background-color: #b1b1b1;
        border-color: #b1b1b1;
        cursor: not-allowed;
        box-shadow: none;
	}

	.text-wrapper {
		font-size: 14px;
        font-weight: 500;
        letter-spacing: 1px;
	}
}
