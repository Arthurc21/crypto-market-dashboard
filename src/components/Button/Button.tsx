import React from 'react'
import './Button.scss'

interface ButtonProps {
	id: string
	text: string
	onClick(event: React.MouseEvent<HTMLButtonElement>): void
	disabled?: boolean
}

export const Button: React.FC<ButtonProps> = ({ id, text, onClick, disabled }: ButtonProps) => {
	return (
		<button
			id={id}
			className="button"
			onClick={(event) => {
				onClick(event)
			}}
			disabled={disabled}
		>
			<span className="text-wrapper">{text}</span>
		</button>
	)
}
