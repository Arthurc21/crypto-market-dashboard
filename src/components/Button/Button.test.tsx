import { create } from "react-test-renderer"
import ReactDOM from "react-dom"
import { Button } from "./Button"

describe ('Button Testing', () => {
    const onClickFunc = jest.fn()
    test('Button snapshot', () => {
        const button = create(
            <Button
                id={'buttonTest'}
                text={'buttonTest'}
                onClick={onClickFunc}
            />
        ).toJSON()
        expect(button).toMatchSnapshot()
    })

    function actionClick(callback: jest.Mock) {
		callback()
	}

    it('Display text correctly and click works', () => {
        const div = document.createElement('div')
        ReactDOM.render(
            <Button
                id={'buttonTest'}
                text={'Button Test'}
                onClick={onClickFunc}
            />,
            div
        )
        actionClick(onClickFunc)
        console.log(div.textContent)
        expect(div.textContent).toEqual('Button Test')
        expect(onClickFunc).toHaveBeenCalled()
        ReactDOM.unmountComponentAtNode(div)
    })
})