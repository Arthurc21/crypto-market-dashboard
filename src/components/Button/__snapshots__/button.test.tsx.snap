// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Testing Button snapshot 1`] = `
<button
  className="button"
  id="buttonTest"
  onClick={[Function]}
>
  <span
    className="text-wrapper"
  >
    buttonTest
  </span>
</button>
`;
